rust/repositories.bzl

```diff
@@ -610,20 +610,20 @@ def rust_repository_set(
         native.register_toolchains(*all_toolchain_names)
         native.register_toolchains(str(Label("//rust/private/dummy_cc_toolchain:dummy_cc_wasm32_toolchain")))
 
-    # Inform users that they should be using the canonical name if it's not detected
-    if "rules_rust" not in native.existing_rules():
-        message = "\n" + ("=" * 79) + "\n"
-        message += (
-            "It appears that you are trying to import rules_rust without using its\n" +
-            "canonical name, \"@rules_rust\" Please change your WORKSPACE file to\n" +
-            "import this repo with `name = \"rules_rust\"` instead."
-        )
-
-        if "io_bazel_rules_rust" in native.existing_rules():
-            message += "\n\n" + (
-                "Note that the previous name of \"@io_bazel_rules_rust\" is deprecated.\n" +
-                "See https://github.com/bazelbuild/rules_rust/issues/499 for context."
-            )
-
-        message += "\n" + ("=" * 79)
-        fail(message)
+    # # Inform users that they should be using the canonical name if it's not detected
```

Comment:
This errors, because the name of the repository is something like `@rules_rust~override`

Reply:
The `native.existing_rules()` API no longer works in Bzlmod. Since we have strict deps in Bzlmod, it's already impossible for the user to depend on something they didn't declare. Maybe we don't need this check anymore?

Reply:
Do we still need this for non-bzlmod usages? Maybe @UebelAndre would know?

If so, I could add a parameter to ignore this check that only is intended to be used by the bzlmod module extension.

Reply:
Sorry, missed this one. The code here can be deleted. I opened https://github.com/bazelbuild/rules_rust/pull/1577 to do so.

Reply:
Thanks!